Azure/sonic-buildimage
# [DELL] S6100 Support PowerCycle in Last Reboot Reason

**- What I did**
Added support to fix power-cycle last reboot cause in S6100
Known Caveat :- If we upgrade old_image (old_image without this fix) to the new_image(with this fix)
If we do a power-cycle instead of fast/warm/cold reset, show reboot-cause will return "Unknown" instead of "power-cycle"
**- How I did it**
Changes Done
modified: files/image_config/process-reboot-cause/process-reboot-cause
modified: files/image_config/platform/rc.local
modified: platform/broadcom/sonic-platform-modules-dell/common/dell_pmc.c
modified: platform/broadcom/sonic-platform-modules-dell/s6000/sonic_platform/chassis.py
modified: platform/broadcom/sonic-platform-modules-dell/s6100/scripts/s6100_platform.sh
modified: platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/chassis.py

rc.local will inform the platform about first boot. Platform Init Script will use the info and update the
logical to return non-hardware in case of first time onie boot/image upgrade.

process-reboot-cause will use the first_boot notification and use it to return Valid last-reset-reason
Currently it will show unkwown which is not correct.
**- How to verify it**
Unit Test Results attached
a) power-cycle the box and issue show reboot-cause
b) Onie install the box and check return value of get_reboot_cause api
c) Upgrade the image and check return value of get_reboot_Cause api

**- Description for the changelog**

[UnitTesting_Final_With_PowerCycle_and_FirstBoot (1).txt](https://github.com/Azure/sonic-buildimage/files/3568038/UnitTesting_Final_With_PowerCycle_and_FirstBoot.1.txt)

[DELL] S6100 Support PowerCycle in Last Reboot Reason
**- A picture of a cute animal (not mandatory but encouraged)**
